### DISCUSSION TITLE
Self hosted runners stuck during setup phase; building action's docker container

### DISCUSSION THREAD

**Discussion by deefdragon** [community/community#66307] [Actions]
### Select Topic Area

Bug

### Body

I have recently encountered an issue with my self hosted runners. This happens _both_ on my VM hosted runner and the K8S deployed runners that I created attempting to remediate. It also happens for multiple different repo's jobs, and multiple different versions of the given action.

When using [this github tag action](https://github.com/anothrNick/github-tag-action) action, during the setup steps, when attempting to build the container for the action, the processes gets stuck.

Specifically, with the following output from the logs, and no furtuer
```
#5 [2/3] RUN apk --no-cache add bash git curl jq && npm install -g semver
#5 sha256:78389989db8fb4f7eafbb87ededa812a7839e1688f02d160fac4a752a583f46d
#5 0.3[57](https://github.com/streemtech/ledger/actions/runs/6106673661/job/16572176367#step:3:57) fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/main/x86_64/APKINDEX.tar.gz
```

building the container on my local desktop (same network) results in  
```
 ---> Running in 6dc8f80dfd7e
fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/community/x86_64/APKINDEX.tar.gz
(1/16) Installing ncurses-terminfo-base (6.4_p20230506-r0)
(2/16) Installing libncursesw (6.4_p20230506-r0)
... (build continues to completion successfully)
```

This only started recently (last few days as far as I can tell). I am at this point unsure what is causing this. The only thing I can think of is it being a bug in my selfhosted runner setup (no idea how to diagnose), or the selfhosted runner itself (given how it just freezes instead of timing out etc.).

**Comment by deefdragon** [comment]
Odd addenda that I didnt notice earlier, but the build steps show up in the logs out of order. Step 4 then 6 then 8 then 7 etc. Im not sure how that might be impacting things.

**Reply by deefdragon** [reply]
Another addenda, This is now showing up only sometimes. (tho more often than not).

**Comment by deefdragon** [comment]
I MAY have solved this with this 
https://github.com/actions/actions-runner-controller/blob/master/TROUBLESHOOTING.md#delay-in-jobs-being-allocated-to-runners
Not sure. Will mark this as the solution if the next 2 days go without freezing.

**Comment by thepieterdc** [comment]
I had a similar issue (job getting stuck when building a container on the fetch alpine apk index phase), and solved it by setting an MTU -- see also https://github.com/earthly/earthly/issues/847
